editoast/editoast_derive/src/error.rs

```diff
@@ -0,0 +1,125 @@
+use std::collections::HashMap;
+
+use darling::{Error, Result};
+use darling::{FromDeriveInput, FromVariant};
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{DataEnum, DeriveInput, Ident, Path};
+
+const DEFAULT_STATUS_CODE: u16 = 400;
+
+#[derive(FromDeriveInput)]
+#[darling(attributes(editoast_error), forward_attrs(allow, doc, cfg))]
+struct ErrorOptions {
+    base_id: String,
```

Comment:
I think that base_id could be an Option<String> since it often is the error name converted to snake_case, which could be the default value.

Reply:
> since it often is the error name converted to snake_case

I don't think it's actually the case. Most of the time we name the errors with the name of the view followed by `Error`.